ci: TestCephBlockPoolController unit test randomly fails

Here is the test:

```
2020-12-02T14:25:16.5472707Z --- FAIL: TestCephBlockPoolController (0.01s)
2020-12-02T14:25:16.5473400Z     controller_test.go:357: 
2020-12-02T14:25:16.5473945Z         	Error Trace:	controller_test.go:357
2020-12-02T14:25:16.5474505Z         	Error:      	Should NOT be empty, but was 
2020-12-02T14:25:16.5475213Z         	Test:       	TestCephBlockPoolController
2020-12-02T14:25:16.5475865Z         	Messages:   	map[]
2020-12-02T14:25:16.5476319Z     controller_test.go:361: 
2020-12-02T14:25:16.5476856Z         	Error Trace:	controller_test.go:361
2020-12-02T14:25:16.5477448Z         	Error:      	Received unexpected error:
2020-12-02T14:25:16.5478118Z         	            	secrets "" not found
2020-12-02T14:25:16.5478782Z         	Test:       	TestCephBlockPoolController
2020-12-02T14:25:16.5479733Z panic: runtime error: invalid memory address or nil pointer dereference [recovered]
2020-12-02T14:25:16.5480594Z 	panic: runtime error: invalid memory address or nil pointer dereference
2020-12-02T14:25:16.5481362Z [signal SIGSEGV: segmentation violation code=0x1 addr=0x120 pc=0x1472927]
2020-12-02T14:25:16.5481792Z 
2020-12-02T14:25:16.5482155Z goroutine 22 [running]:
2020-12-02T14:25:16.5482674Z testing.tRunner.func1(0xc0004fca00)
2020-12-02T14:25:16.5483506Z 	/opt/hostedtoolcache/go/1.13.15/x64/src/testing/testing.go:874 +0x3a3
2020-12-02T14:25:16.5484073Z panic(0x15da880, 0x25ec3d0)
2020-12-02T14:25:16.5484679Z 	/opt/hostedtoolcache/go/1.13.15/x64/src/runtime/panic.go:679 +0x1b2
2020-12-02T14:25:16.5485643Z github.com/rook/rook/pkg/operator/ceph/pool.TestCephBlockPoolController(0xc0004fca00)
2020-12-02T14:25:16.5486752Z 	/home/runner/go/src/github.com/rook/rook/pkg/operator/ceph/pool/controller_test.go:362 +0x1a27
2020-12-02T14:25:16.5487487Z testing.tRunner(0xc0004fca00, 0x186a2f8)
2020-12-02T14:25:16.5488166Z 	/opt/hostedtoolcache/go/1.13.15/x64/src/testing/testing.go:909 +0xc9
2020-12-02T14:25:16.5488754Z created by testing.(*T).Run
2020-12-02T14:25:16.5489381Z 	/opt/hostedtoolcache/go/1.13.15/x64/src/testing/testing.go:960 +0x350
2020-12-02T14:25:16.5490057Z FAIL	github.com/rook/rook/pkg/operator/ceph/pool	0.031s
```

--- a/pkg/operator/ceph/pool/controller_test.go
+++ b/pkg/operator/ceph/pool/controller_test.go
@@ -348,19 +348,27 @@ func TestCephBlockPoolController(t *testing.T) {
 	pool.Spec.Mirroring.Mode = "image"
 	err = r.client.Update(context.TODO(), pool)
 	assert.NoError(t, err)
-	res, err = r.Reconcile(req)
-	assert.NoError(t, err)
-	assert.False(t, res.Requeue)
-	err = r.client.Get(context.TODO(), req.NamespacedName, pool)
-	assert.NoError(t, err)
-	assert.Equal(t, cephv1.ConditionReady, pool.Status.Phase)
+	for i := 0; i < 5; i++ {
+		res, err = r.Reconcile(req)
+		assert.NoError(t, err)
+		assert.False(t, res.Requeue)
+		err = r.client.Get(context.TODO(), req.NamespacedName, pool)
+		assert.NoError(t, err)
+		assert.Equal(t, cephv1.ConditionReady, pool.Status.Phase)
+		if _, ok := pool.Status.Info[RBDMirrorBootstrapPeerSecretName]; ok {
+			break
+		}
+		logger.Infof("FIX: trying again to update the mirroring status")
+	}
 	assert.NotEmpty(t, pool.Status.Info[RBDMirrorBootstrapPeerSecretName], pool.Status.Info)
 
 	// fetch the secret
 	myPeerSecret, err := c.Clientset.CoreV1().Secrets(namespace).Get(ctx, pool.Status.Info[RBDMirrorBootstrapPeerSecretName], metav1.GetOptions{})
 	assert.NoError(t, err)
-	assert.NotEmpty(t, myPeerSecret.Data["token"], myPeerSecret.Data)
-	assert.NotEmpty(t, myPeerSecret.Data["pool"])
+	if myPeerSecret != nil {
+		assert.NotEmpty(t, myPeerSecret.Data["token"], myPeerSecret.Data)
+		assert.NotEmpty(t, myPeerSecret.Data["pool"])
+	}
 }
 
 func TestConfigureRBDStats(t *testing.T) {

